src/DetectorConstruction.cpp

```diff
@@ -105,7 +106,7 @@ G4VPhysicalVolume* DetectorConstruction::Construct()
   }
 
   // DETECTOR: separate class
-  m_energyCounter = new EnergyCounter( "Detector", m_decayTimeFinder, m_outputFileName );
+  m_energyCounter = new EnergyCounter( "Detector", m_decayTimeFinder, m_outputFileName, m_decayOutputFileName );
```

Comment:
Right, in this sense it may not be the best place for that but what other place would you suggest? This just seems like the easiest solution...

Reply:
I would have just had the decay finder read it out directly. Since it's information about the decay, it makes sense to put this code in the same place.

Reply:
I moved creating the csv file to DecayTimeFinderAction as you suggested